test/helpers/getCodeFromSass.js

```diff
@@ -56,6 +56,27 @@ async function getCodeFromSass(testId, options) {
     testFolder,
     "node_modules/package-with-exports/style.scss"
   );
+  const pathToSassPackageWithExportsFieldsAndCustomConditionReplacer = () => {
+    if (context.packageExportsCustomConditionTestVariant === 1) {
+      return path.resolve(
+        testFolder,
+        "node_modules/package-with-exports-and-custom-condition/style-1.scss"
+      );
+    }
+
+    if (context.packageExportsCustomConditionTestVariant === 2) {
+      return path.resolve(
+        testFolder,
+        "node_modules/package-with-exports-and-custom-condition/style-2.scss"
+      );
+    }
+
+    console.warn(
+      "Expedted to receive .packageExportsCustomConditionTestVariant to properly resolve stylesheet in sass only compilation. "
+    );
+    return "";
+  };
```

Comment:
There wasn't a great pattern to write the type of tests that made sense here so I had to introduce some new constructs (such as this `context` param) to tell `getCodeFromSass` how to properly resolve the file that the tests need to compare against (based on the custom condition).